### PR TITLE
Add remove peer logic for damaged regions when handling store heartbeat

### LINKED ISSUE #4868: Add remove peer logic for damaged regions when handling store heartbeat
### Describe your feature request related problem

ref https://github.com/tikv/tikv/issues/10578

### Describe the feature you'd like

To support sst recovery. In short, PD need remove damaged peer when receiving store heartbeat.

### Describe alternatives you've considered

none

### Teachability, Documentation, Adoption, Migration Strategy

ref https://github.com/tikv/tikv/issues/10578

### PR DESCRIPTION
**Pull request by hunterlxt**
Signed-off-by: Xintao <hunterlxt@live.com>

### What problem does this PR solve?

Issue Number: close #4868 

### What is changed and how it works?

```commit-message
Add remove peer logic for damaged regions when handling store heartbeat
```

### Check List

Tests

- Unit test
- Integration test

Code changes

- only logic changed

Side effects

- when this kind of errors happen, user need intervention manually with pd-ctl

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
Add remove peer logic for damaged regions when handling store heartbeat
```

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -1925,20 +1923,29 @@ func (s *GrpcServer) sendAllGlobalConfig(ctx context.Context, server pdpb.PD_Wat
...
-	return s.GetHandler().AddEvictOrGrant(float64(stats.GetStoreId()), schedulers.EvictLeaderName)
+	for _, regionID := range stats.GetDamagedRegionsId() {
+		// Remove peers to make sst recovery physically delete files in TiKV.
+		err := s.GetHandler().AddRemovePeerOperator(regionID, stats.GetStoreId())
+		if err != nil {
+			log.Error("store damaged but can't add remove peer operator",
+				zap.Uint64("region-id", regionID), zap.Uint64("store-id", stats.GetStoreId()), zap.String("error", err.Error()))
+		} else {
+			log.Warn("added remove peer operator due to damaged region",
+				zap.Uint64("region-id", regionID), zap.Uint64("store-id", stats.GetStoreId()), zap.String("error", err.Error()))
```
Comment (rleungx) [inline server/grpc_service.go:1946]: no need error

**Review by rleungx — APPROVED** [review]
The rest LGTM

File: server/grpc_service.go (hunk 1 above)
Comment (hunterlxt) [reply server/grpc_service.go:1946]: no need warn?

**Review by lhy1024 — APPROVED** [review]

File: server/grpc_service.go (hunk 1 above)
Comment (rleungx) [reply server/grpc_service.go:1946]: I mean the error field will be nil

**Pull request merged**